api/src/main/java/ai/djl/nn/Block.java

```diff
@@ -127,6 +128,8 @@ default NDList forward(ParameterStore parameterStore, NDList inputs, boolean tra
         return forward(parameterStore, inputs, training, null);
     }
 
+    NativeResource<Long> forward(NativeResource<Long>[] inputs);
```

Comment:
Instead of adding these additions that may not apply to most of the `Block`s, can you instead modify `GPT2PtLMAdapter` to cast from `Block` to `PtSymbolBlock` before making the `IValue` forward call?